std/utf.d

```diff
@@ -3175,6 +3189,31 @@ auto ref byCodeUnit(R)(R r)
     return r;
 }
 
+///
+unittest
+{
+    auto r = "Hello, World!".byCodeUnit();
+    static assert(hasSlicing!(typeof(r)));
+    static assert(isRandomAccessRange!(typeof(r)));
+    static assert(is(ElementType!(typeof(r)) == immutable char));
+
+    // contrast with the range capabilities of standard strings
+    auto s = "Hello, World!";
+    static assert(isBidirectionalRange!(typeof(r)));
+    static assert(is(ElementType!(typeof(s)) == dchar));
+
+    static assert(!isRandomAccessRange!(typeof(s)));
+    static assert(!hasSlicing!(typeof(s)));
+}
+
+/// `byCodeUnit` does no Unicode encoding
+unittest
+{
+    auto s = "noe\u0308l".byCodeUnit; // noël using e + combining diaeresis
+    s.popFrontN(2);
+    assert(s.front == 'e'); // not equal to ë
```

Comment:
I still think this should also show that "noël" with a one-code-point ë isn't safe either.

```d
    auto noel1 = "noe\u0308l"; // noël using e + combining diaeresis
    assert(noel1.byCodeUnit[2] != 'ë');
    assert(noel1.byCodeUnit[2] == 'e');
    
    auto noel2 = "no\u00EBl"; // noël using a precomposed ë character
    assert(noel2.byCodeUnit[2] != 'ë'); /* The code unit at index 2 is just the
                                        first of a sequence that encodes 'ë'. */
```